### DISCUSSION TITLE
Deviation score across clusters for handpicked list of TFs

### DISCUSSION THREAD

**Discussion by Dearyl14** [GreenleafLab/ArchR#2019] [Questions / Documentation]
Dear ArchR team and userbase,

I have been trying to create the Deviation Score heatmaps from my own data as presented in figure 4F of Satpathy& Granja et al. (Nature Biotech, 2019) and as recently shown in Figure 5 of Hickey et al. (Nature, 2023). 

Having generated deviation scores across single cells, I am having trouble extracting a score for each cluster for a designated set of TFs (i.e. known TFs important for the given lineage). Is there an easy way in ArchR to output the cluster specific deviation scores for a user-defined list of TF motifs?

I have been able to get the deviation scores for positively regulated TFs, but those don't distinguish by cluster.

Sorry if I am missing something simple here, but any hint towards which functions and parameters to use would be much appreciated. Or maybe these heatmaps are done by hand from the single cell deviation scores?

Thanks for your time,
Luke

**Comment by rcorces** [comment]
This information is stored in the Deviations matrix which can be accessed with `getMatrixFromProject()`

**Reply by Dearyl14** [reply]
Thanks, Ryan! We figured this out and made some nice motif activity heatmaps.
